src/Pillar-ExporterCore/PRTarget.class.st

```diff
@@ -8,15 +8,27 @@ Class {
 	#category : #'Pillar-ExporterCore-Base'
 }
 
+{ #category : #'as yet unclassified' }
+PRTarget class >> associateTarget: aFormat [
+	^ self allSubclasses
+		select: [ :each | (each isAbstract not) and: each builderName = aFormat asString ]
```

Comment:
the `and: ` method uses typically blocks as arguments.
For example

```smalltalk
bla and: [ blu ]
```

Reply:
Also

 - `associateTarget:` is a strange name, it does not follow any of the conventions, and it is not clear what it does. Maybe it could be renamed as `targetByFormatName:`?
- I don't think that this method should return a collection. There are several reasons. First, the name of the method does not imply it returns a collection. Second, you're using a collection to have the posibility to "return nothing".

There are other ways to solve this :).

One possibility is to use `detect:`. The `detect:` will find an object with the condition and return it. If many satisfy the condition it returns the first. If none satisfy the condition, an exception is raised.

In addition, there is `detect:ifNone:` which receives also a block to handle the situation where no element was found.

This means you could for example define `targetByFormatName:` that fails if none is found. And: `targetByFormatName:ifNone:` that receives a block. Then your code will use the second I think.

Another solution is to just return `nil`. But I don't like that because you force all the users of your method to check the result against `nil`, even when most of the times it is not necessary. And that makes the code unreadable super fast.

Reply:
1. I edited introducing a block and and: :)
2. When no target is found, this method returns an empty ordered collection so that it always returns the same type of object. That was my first thought. But we can think of `detect: ifNone:` using in None case, the PRInvalidTarget :)

Reply:
Solved so!